core/encoding/json/unmarshal.odin

```diff
@@ -86,6 +86,12 @@ assign_bool :: proc(val: any, b: bool) -> bool {
 @(private)
 assign_int :: proc(val: any, i: $T) -> bool {
 	v := reflect.any_core(val)
+	ti := runtime.type_info_base(type_info_of(v.id))
+
+	if bs, ok := ti.variant.(runtime.Type_Info_Bit_Set); ok {
+			v.id = reflect.type_info_core(bs.elem).id
+	}
+
```

Comment:
I am not sure if this is a good solution or not. Is it better to save it as an integer or to save it as an array of values?

Reply:
If saved as an array of values, then it would keep working if you later renumber the enum's iota.

Reply:
I'm not sure I understand how an array of values will keep the unmarshalling consistent if the enum was reordered or renumbered. Can you give an example?

Reply:
```cpp
Enum :: enum{.A, .B, .C, .D, .E}
Bit_Set :: bit_set[Enum;u8]
v := Bit_Set{.A, .C, .E}
```
So far it's just bit 0, 2, and 4 set, right, so we'd encode it as 1 + 4 + 16 = 21.
Now unmarshal after we do this:
```cpp
Enum :: enum{.A, .A2, .B, .B2, .C, .D, .E}
```
A C E in -> 21 -> A B C out.

Now marshal `{.A, .C, .E}` as `"A,C,E"`. Rearrange the enum to `{.E, .F, .A, .C, .B}`, but `"A,C,E"` still unmarshals to the same `{.A, .C, .E}`.

Reply:
And better to get an unmarshal error because a value "Foo" is missing in the enum since you marshaled it, then to unmarshal A = 5 to Z = 5 later without being able to notice it was wrong.

Reply:
Right, I see what you mean now, I thought an array of values meant an array of 0s and 1s. That does seem better than an integer.
Should this get merged first and then we can change bit_set (un)marshalling to be an array or should I close this PR?